src/Git/Git.cpp

```diff
@@ -988,9 +985,47 @@ CString CGit::GetLogCmd(const CString &range, const CTGitPath *path, int count,
 
 	CString st1,st2;
 
-	if( Filter && (Filter->m_From != -1))
+	if (Filter && Filter->m_NumberOfLogs > 0 && Filter->m_NumberOfLogsScale == CFilterData::SHOW_LAST_N_COMMITS)
+		num.Format(_T("-n%d"), Filter->m_NumberOfLogs);
+
+	if (Filter)
 	{
-		st1.Format(_T(" --max-age=%I64u "), Filter->m_From);
+		if (Filter->m_NumberOfLogsScale == CFilterData::SHOW_LAST_SELECTED_DATE)
+		{
+			CString key = _T("Software\\TortoiseGit\\History\\LogDlg_FromDate\\") + g_Git.m_CurrentDir + _T("\\FromDate0");
+			key.Replace(_T(':'), _T('_'));
+			CString str = CRegString(key, _T("-1"));
+			Filter->m_From = (__time64_t)(_tcstoui64(str.GetBuffer(0), nullptr, 10));
+			str.ReleaseBuffer();
+			if (Filter->m_From > 0)
+				st1.Format(_T(" --max-age=%I64u "), Filter->m_From);

```

Comment:
Not sure line 995~999.
Should they move outside of this function?
(Put them in GitLogListBase.cpp?)